services/github/src/handlers.js

```diff
@@ -504,17 +490,47 @@ async function statusHandler(message) {
     `Attempting to update status of the checkrun for ${organization}/${repository}@${sha} (${taskState.conclusion})`,
   );
   try {
+    const taskDefinition = await this.queueClient.task(taskId);
+    const {customCheckRun} = (taskDefinition.extra && taskDefinition.extra.github) || {};
+    const {textArtifactName} = customCheckRun || CUSTOM_CHECKRUN_TEXT_ARTIFACT_NAME;
+
+    let customCheckRunText = '';
+
+    const url = this.queueClient
+      .buildUrl(this.queueClient.getArtifact, taskId, runId, textArtifactName);
+
+    let res = await utils.throttleRequest({url, method: 'GET'});
+
+    if (res.status >= 400 && res.status !== 404) {
+      await this.createExceptionComment({
+        debug,
+        instGithub,
+        organization,
+        repository,
+        sha,
+        error: res.response.error,
+      });
+
+      if (res.status < 500) {
+        await this.monitor.reportError(res.response.error);
+      }
+    } else if (res.status >= 200 && res.status < 300) {
+      customCheckRunText = res.text.toString();
+    }
+
     if (checkRun) {
       await instGithub.checks.update({
         ...taskState,
         owner: organization,
         repo: repository,
         check_run_id: checkRun.checkRunId,
+        output: {
```

Comment:
It would probably make sense to move at least `output` outside the `if`, so the text construction isn't duplicated.